source/base/parameter_handler.cc

```diff
@@ -693,21 +836,47 @@ namespace Patterns
 
 
 
-  std::string Map::description () const
+  std::string Map::description (const OutputStyle style) const
   {
-    std::ostringstream description;
-
-    description << description_init
-                << " map of <"
-                << key_pattern->description() << ":"
-                << value_pattern->description() << ">"
-                << " of length " << min_elements << "..." << max_elements
-                << " (inclusive)";
-    if (separator != ",")
-      description << " separated by <" << separator << ">";
-    description << "]";
-
-    return description.str();
+    switch (style)
+      {
+      case Machine:
+      {
+        std::ostringstream description;
+
+        description << description_init
+                    << " map of <"
+                    << key_pattern->description(style) << ":"
+                    << value_pattern->description(style) << ">"
+                    << " of length " << min_elements << "..." << max_elements
+                    << " (inclusive)";
+        if (separator != ",")
+          description << " separated by <" << separator << ">";
+        description << "]";
+
+        return description.str();
+      }
+      case Text:
+      case LaTeX:
+      {
+        std::ostringstream description;
+
+        description << "a key-value map of "
+                    << min_elements << " to " << max_elements
+                    << " elements ";
+        if (separator != ",")
+          description << " separated by <" << separator << "> ";
+        description << " where the each key is " << key_pattern->description(style)
```

Comment:
the each -> each

Same thing here with bracketing the inner description somehow.